packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -680,26 +838,52 @@ abstract class State<T extends StatefulWidget> {
     assert(() { _debugLifecycleState = _StateLifecycle.defunct; return true; });
   }
 
-  /// Returns another Widget out of which this [StatefulWidget] is built.
-  /// Typically that Widget will have been configured with further children,
-  /// such that really this function returns a tree of configuration.
+  /// Describes the part of the user interface represented by this widget.
+  ///
+  /// The framework calls this method in a number of different situations:
   ///
-  /// The given build context object contains information about the location in
-  /// the tree at which this widget is being built. For example, the context
-  /// provides the set of inherited widgets for this location in the tree.
+  ///  * After calling [initState].
+  ///  * After calling [didUpdateConfig].
+  ///  * After receiving a call to [setState].
+  ///  * After a dependencies of this [State] objects changes (e.g., an
+  ///    [InheritedWidget] referenced by the previous [build] changes).
+  ///  * After calling [deactivate] and then reinserting the [State] object into
+  ///    the tree at another location.
   ///
-  /// The context argument is always the same as [State.context]. This argument
-  /// is provided redundantly here to match the [WidgetBuilder] function
-  /// signature used by [StatelessWidget.build] and other widgets.
+  /// The framework replaces the subtree below this widget with the widget
+  /// returned by this method, either by updating the existing subtree or by
+  /// removing the subtree and inflating a new subtree, depending on whether the
+  /// widget returned by this method [Widget.canUpdate] the root of the
```

Comment:
"...the widget returned by this method Widget.canUpdate the root of the existing subtree" is confusing.

Suggest: "the widget returned by this method can update the root of the existing subtree, as determined by calling Widget.canupdate".